### PR TITLE
fix#11375: suggestions improvement for quick filter dropdown options

### LINKED ISSUE #11375: Improve suggestions for explore quick dropdown filters
**Affected module**
Does it impact the UI, backend or Ingestion Framework?
UI, backend

**Describe the bug**

- [ ] Use only aggregate API for showing the suggestion options for explore page quick dropdown filters. ( Currently, 2 APIs are being used for fetching the options, `Aggregates` API for getting options without any search text and `Suggest` API for with search text)

**To Reproduce**

Screenshots or steps to reproduce

**Expected behavior**

**Version:**
 - Python version:
 - OpenMetadata version: [e.g. 0.8]
 - OpenMetadata Ingestion package version: [e.g. `openmetadata-ingestion[docker]==XYZ`]

**Additional context**

### PR DESCRIPTION
**Pull request by aniketkatkar97**
### Describe your changes:

Fixes #11375 

I worked on updating the logic to get the options for advanced search quick filters as well as advanced search modal suggestions options on explore page

#
### Type of change:

- [ ] Bug fix
- [ ] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/cypress/common/advancedSearch.js [hunk 1]
```diff
@@ -334,20 +350,36 @@ export const addTag = (tag) => {
...
-    .contains(tag);
+
+  tagsArray.map((tagName) => {
+    cy.get('[data-testid="entity-tags"]')
+      .scrollIntoView()
+      .should('be.visible')
+      .contains(tagName);
+  });
+
+  cy.log('exit');
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/cypress/common/advancedSearch.js:382]: log?

File: openmetadata-ui/src/main/resources/ui/cypress/common/advancedSearch.js [hunk 2]
```diff
@@ -392,13 +424,11 @@ export const checkAddGroupWithOperator = (
         .should('be.visible')
         .type(searchCriteria_1);
     } else {
-      interceptURL('GET', '/api/v1/search/suggest?q=*', 'suggestApi');
       cy.get('.widget--widget > .ant-select > .ant-select-selector')
         .eq(index_1)
         .should('be.visible')
         .type(searchCriteria_1);
 
-      verifyResponseStatusCode('@suggestApi', 200);
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/cypress/common/advancedSearch.js:401]: Don't we need to intercept API whichever we are using to search?

File: openmetadata-ui/src/main/resources/ui/cypress/common/advancedSearchQuickFilters.js [hunk 3]
```diff
@@ -38,22 +38,12 @@ export const searchAndClickOnOption = (
...
-    'suggestAPI'
-  );
-
   cy.get('[data-testid="search-input"]')
     .should('exist')
     .and('be.visible')
     .clear()
     .type(optionName);
 
-  verifyResponseStatusCode('@suggestAPI', 200);
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/cypress/common/advancedSearchQuickFilters.js:55]: same

File: openmetadata-ui/src/main/resources/ui/cypress/common/advancedSearch.js (hunk 1 above)
Comment (aniketkatkar97) [reply openmetadata-ui/src/main/resources/ui/cypress/common/advancedSearch.js:382]: Sorry, I missed that. I'll take care of it

File: openmetadata-ui/src/main/resources/ui/cypress/common/advancedSearch.js (hunk 2 above)
Comment (aniketkatkar97) [reply openmetadata-ui/src/main/resources/ui/cypress/common/advancedSearch.js:401]: We added this wait as we used the suggest API to get options after searching. But now we are searching internally within the results we get from the Aggregates API, so there is no call for getting suggestions

**Review by pmbrull — APPROVED** [review]

**Review by chirag-madlani — APPROVED** [review]

File: openmetadata-ui/src/main/resources/ui/cypress/common/advancedSearch.js [hunk 4]
```diff
@@ -297,11 +299,18 @@ export const addOwner = (searchTerm, ownerName) => {
     });
 };
 
-export const addTier = (tier) => {
+export const addTier = (
+  tier,
+  entityName,
+  ServiceName,
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/cypress/common/advancedSearch.js:305]:
```suggestion
  serviceName,
```

**Review by Sachin-chaurasiya — COMMENTED** [review]
Left a few comments, thanks

**Comment by aniketkatkar97** [comment]
As the behaviour of the quick dropdown has been refactored, we don't need these changes anymore.

**Pull request closed without merging**